doc/man3/EVP_DigestInit.pod

```diff
@@ -121,16 +128,29 @@ EVP_MD_CTX_ctrl() must be called after EVP_DigestInit_ex(). Other restrictions
 may apply depending on the control type and digest implementation.
 See L</CONTROLS> below for more information.
 
-=item EVP_MD_CTX_get_params
+=item EVP_MD_get_params()
+
+Retrieves the requested list of B<params> from a MD B<md>.
+See L</PARAMS> below for more information.
+
+=item EVP_MD_CTX_get_params()
 
 Retrieves the requested list of B<params> from a MD context B<ctx>.
 See L</PARAMS> below for more information.
 
-=item EVP_MD_CTX_set_params
+=item EVP_MD_CTX_set_params()
 
-Sets the list of <params> into a MD context B<ctx>.
+Sets the list of B<params> into a MD context B<ctx>.
 See L</PARAMS> below for more information.
 
+=item EVP_MD_get_param_types(), EVP_MD_CTX_get_param_types(),
+EVP_MD_CTX_set_param_types()
+
+Get a B<OSSL_PARAM> array that describes the retrievable and settable
+parameters, i.e. parameters that can be used with EVP_MD_get_params(),
+EVP_MD_CTX_get_params() and EVP_MD_CTX_set_params(), respectively.
```

Comment:
So this is kind of confusing. If I'm reading this correctly `EVP_MD_CTX_set_param_types` is actually a "getter" to get the param types for `EVP_MD_CTX_set_params`. That seems wrong somehow.

Reply:
The alternative is names like `EVP_MD_CTX_get_set_param_types`, and even worse, `EVP_MD_CTX_get_get_param_types`. I find that unacceptable. If you have a better idea, I'm all ears.

Reply:
I'd prefer `EVP_MD_CTX_get_settable_types` and `EVP_MD_CTX_get_gettable_types`. But I do think we cannot just call it `EVP_MD_CTX_set_param_types`

Reply:
Or another option: `EVP_MD_CTX_get_types_set_params`

Reply:
Settable and gettable types... Yes, I'm buying! Will do the change accross the board this even (that includes #8877)

Reply:
The get/set nomenclature is confusing.  params.h uses them in one way and this in the opposite.  I agree with settable and gettable here and still want to rename the params.h functions to avoid misunderstandings.

Reply:
It would be even possible to drop the get - i.e. EVP_MD_CTX_settable_types() EVP_MD_CTX_gettable_types() - as there will be no set function, the get is almost redundant.

Reply:
Done